modules/ballerina-test/src/test/java/org/ballerinalang/test/types/map/BMapValueTest.java

```diff
@@ -184,4 +186,47 @@ public void testBMapToString() {
         
         Assert.assertEquals(map.stringValue(), "{\"key1\":1, \"key2\":\"foo\", \"key3\":<bar>hello</bar>}");
     }
+
+    @Test(dependsOnMethods = "testGrammar")
+    public void testMapOrder() {
+//        BValue[] returnVals = BLangFunctions.invokeNew(programFile, "testMapOrder", new BValue[0]);
+        BValue[] returnVals = BRunUtil.invoke(programFile, "testMapOrder", new BValue[0]);
+        BMap m = (BMap) returnVals[0];
+        Set set = m.keySet();
+        Iterator i = set.iterator();
+        int counter = 0;
+        String[] values = {"Element 1", "Element 2", "Element 3"};
+        while (i.hasNext()) {
```

Comment:
For testing map order, need to check both the key and value also, that is loop through all the entries of the map. This can be done with the for-each loop, for iterating through the entries, and check both the key and value for each entry, if they are correct. Avoid using iterators when the for-each loop is sufficient.